src/orbits/kepler.jl

```diff
@@ -1,110 +1,230 @@
+        r = orbit.aRₛ
     else
-        r = -orbit.a * (1 - orbit.ecc^2) / (1 + orbit.ecc * cosf)
+        r = orbit.aRₛ * (1 - orbit.ecc^2) / (1 + orbit.ecc * cosν)
     end
-    return rotate_vector(orbit, r * cosf, r * sinf)
+    return rotate_vector(orbit, r * cosν, r * sinν)
 end
 
-function get_true_anomaly(orbit::KeplerianOrbit, t)
-    M = orbit.n * ((t - orbit.t0) - orbit.tref)
-    return kepler(M, orbit.ecc)
+function get_M(orbit::KeplerianOrbit, t)
+    if orbit.ecc == 0.0
+        M = M₀ + orbit.n * (t - orbit.t₀) / 2.0
+    else
+        sinω, cosω = sincos(orbit.ω)
+        E₀ = 2.0 * atan(
+            sqrt(1.0 - orbit.ecc) * cosω,
+            sqrt(1.0 + orbit.ecc) * (1.0 + sinω),
+        )
+        M₀_ecc = E₀ - orbit.ecc * sin(E₀)
+        M = M₀_ecc + orbit.n * (t - orbit.t₀)
+    end
+    return M
 end
 
-function kepler(M, ecc)
-    E = kepler_solver(M, ecc)
-    return sincos(E)
+# Returns sin(ν), cos(ν)
+function get_true_anomaly(orbit::KeplerianOrbit, t)
+    #M = get_M(orbit, t)
+    M = M₀ + orbit.n * (t - orbit.t₀) / 2.0
+    E = kepler_solver(M, orbit.ecc)
+    return sincos(trueanom(E, orbit.ecc))
 end
-kepler(M, ::Nothing) = sincos(M)
 
+# Transform from orbital plane to equatorial plane
 function rotate_vector(orbit::KeplerianOrbit, x, y)
-    # rotate about z0 axis by ω
-    if orbit.ecc === nothing
+    sini, cosi = sincos(orbit.incl)
+    sinΩ, cosΩ = sincos(orbit.Ω)
+    sinω, cosω = sincos(orbit.ω)
+
+    # Rotate about z0 axis by ω
+    if orbit.ecc == 0.0
         x1, y1 = x, y
     else
-        # x1 = orbit.cosω * x - orbit.sinω * y
-        # y1 = orbit.sinω * x + orbit.cosω * y
+        x1 = cosω * x - sinω * y
+        y1 = sinω * x + cosω * y
     end
 
-    # rotate about x1 axis by -incl
+    # Rotate about x1 axis by -incl
     x2 = x1
-    y2 = orbit.cos_incl * y1
-    Z = -orbit.sin_incl * y1
-
-    return SA[x2, y2, Z]
-
-    # rotate about z2 axis by Ω
-    # if orbit.Ω === nothing
-    #     return SA[x2, y2, Z]
-    # end
-    # X = orbit.cosΩ * x2 - orbit.sinΩ * y2
-    # Y = orbit.sinΩ * x2 + orbit.cosΩ * y2
-    # return SA[X, Y, Z]
-end
\ No newline at end of file
+    y2 = cosi * y1
+    Z = -sini * y1
+
+    # Rotate about z2 axis by Ω
+    X = cosΩ * x2 - sinΩ * y2
+    Y = sinΩ * x2 + cosΩ * y2
+
+    return SA[X, Y, Z]
+end
+
+function Base.show(io::IO, orbit::KeplerianOrbit)
+    a = orbit.a
+    aRₛ = orbit.aRₛ
+    b = orbit.b
+    ecc = orbit.ecc
+    P = orbit.P
+    ρₛ = orbit.ρₛ
+    Rₛ = orbit.Rₛ
+    t₀ = orbit.t₀
+    incl = orbit.incl
+    Ω = orbit.Ω
+    ω = orbit.ω
+    print(
+        io,
+        """KeplerianOrbit(
+            a=$(upreferred(orbit.a)), aRₛ=$(orbit.aRₛ),
+            b=$(orbit.b), ecc=$(orbit.ecc), P=$(orbit.P),
+            ρₛ=$(orbit.ρₛ), Rₛ=$(orbit.Rₛ),
+            t₀=$(orbit.t₀), incl=$(orbit.incl),
+            Ω=$(orbit.Ω), ω = $(orbit.ω)
+        )"""
+    )
+end
+
+function Base.show(io::IO, ::MIME"text/plain", orbit::KeplerianOrbit)
```

Comment:
print an animation in unicode of the top-down orbit? (only kidding 😄)

Reply:
That'd be _cool_!

Reply:
Lasers or we riot

Reply:
Render it to a stack of image arrays and play it back in the terminal with https://github.com/IanButterworth/VideoInTerminal.jl

(@giordano pinged me 🙂 )

Reply:
Hello fellow Ian. This looks amazing!

Reply:
Hey Ian. btw, it looks significantly better on linux than the MacOS examples in the readme

Reply:
Oh I am totally showing off the webcam feature to some people